lib/src/main/java/graphql/nadel/engine/instrumentation/NadelInstrumentationTimer.kt

```diff
@@ -0,0 +1,100 @@
+package graphql.nadel.enginekt.instrumentation
+
+import graphql.execution.instrumentation.InstrumentationState
+import graphql.nadel.instrumentation.NadelInstrumentation
+import graphql.nadel.instrumentation.parameters.NadelInstrumentationTimingParameters
+import graphql.nadel.instrumentation.parameters.NadelInstrumentationTimingParameters.Step
+import java.time.Duration
+
+internal class NadelInstrumentationTimer(
```

Comment:
This is a complicated class - especially the `Batch` timer code - but there does not seem to be any unit tests on it.

I think there must be